clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/agent/KubernetesCachingAgent.java

```diff
@@ -72,31 +79,87 @@
 
   @Getter protected final Long agentInterval;
 
+  protected final KubernetesConfigurationProperties configurationProperties;
+
+  protected final KubernetesSpinnakerKindMap kubernetesSpinnakerKindMap;
+
   protected KubernetesCachingAgent(
       KubernetesNamedAccountCredentials namedAccountCredentials,
       ObjectMapper objectMapper,
       Registry registry,
       int agentIndex,
       int agentCount,
-      Long agentInterval) {
+      Long agentInterval,
+      KubernetesConfigurationProperties configurationProperties,
+      KubernetesSpinnakerKindMap kubernetesSpinnakerKindMap) {
     this.accountName = namedAccountCredentials.getName();
     this.credentials = namedAccountCredentials.getCredentials();
     this.objectMapper = objectMapper;
     this.registry = registry;
     this.agentIndex = agentIndex;
     this.agentCount = agentCount;
     this.agentInterval = agentInterval;
+    this.configurationProperties = configurationProperties;
+    this.kubernetesSpinnakerKindMap = kubernetesSpinnakerKindMap;
   }
 
   protected Map<String, Object> defaultIntrospectionDetails() {
     Map<String, Object> result = new HashMap<>();
     result.put("namespaces", getNamespaces());
-    result.put("kinds", primaryKinds());
+    result.put("kinds", kindsToCache());
     return result;
   }
 
   protected abstract List<KubernetesKind> primaryKinds();
 
+  protected abstract boolean cachesKind(KubernetesKind kind);
+
+  protected List<KubernetesKind> kindsToCache() {
+    List<KubernetesKind> cacheKinds;
+
+    if (configurationProperties.getCache().isCacheAll()) {
+      cacheKinds = primaryKinds();
+
+    } else if (configurationProperties.getCache().getCacheKinds() != null
+        && configurationProperties.getCache().getCacheKinds().size() > 0) {
```

Comment:
If someone wants to cache nothing, how do they do it?  Seems like an empty list ends up caching SPINNAKER_UI_KINDS at the moment.

Reply:
There may be a flag to cache nothing for all cloud providers, but I could see wanting to cache ec2 accounts, but not kubernetes accounts (i.e. if the infrastructure tabs are disabled in the UI).

Reply:
Thanks for adding `kubernetes.cache.enabled`.